research/delf/delf/python/training/global_features/train.py

```diff
@@ -0,0 +1,719 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Training script for Global Features model."""
+
+import math
+import os
+import time
+
+import numpy as np
+import pickle
+import tensorflow as tf
+import tensorflow_addons as tfa
+
+from absl import flags
+from absl import app
+from absl import logging
+
+from delf.python.datasets import tuples_dataset
+from delf.python.datasets.revisited_op import dataset as testdataset
+from delf.python.training.losses import ranking_losses
+from delf.python.training import global_features_utils
+from delf.python.training import tensorboard_utils
+from delf.python.training.model import global_model
+from delf.python.datasets.sfm120k import sfm120k
+from delf.python import whiten
+
+_TRAINING_DATASET_NAMES = ['retrieval-SfM-120k']
+_TEST_DATASET_NAMES = ['roxford5k', 'rparis6k']
+_PRECOMPUTE_WHITEN_NAMES = ['retrieval-SfM-30k', 'retrieval-SfM-120k']
+_POOL_NAMES = ['mac', 'spoc', 'gem']
+_LOSS_NAMES = ['contrastive', 'triplet']
+_OPTIMIZER_NAMES = ['sgd', 'adam']
+_MODEL_NAMES = global_features_utils.get_standard_keras_models()
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_boolean('debug', default=False, help='Debug mode.')
+
+# Export directory, training and val datasets, test datasets.
+flags.DEFINE_string('data_root', default="data", help='Path to the data.')
+flags.DEFINE_string('directory', default="data",
+                    help='Destination where trained network should be saved.')
+flags.DEFINE_enum('training_dataset', default='retrieval-SfM-120k',
+                  enum_values=_TRAINING_DATASET_NAMES,
+                  help='Training dataset: ' +
+                       ' | '.join(_TRAINING_DATASET_NAMES) +
+                       ' (default: retrieval-SfM-120k).')
+flags.DEFINE_bool('val', default=True, help='Whether to run validation.')
+flags.DEFINE_bool('val_eccv2020', default=False,
+                  help='New validation dataset used with ECCV 2020 paper.')
+flags.DEFINE_string('test_datasets', default='roxford5k,rparis6k',
+                    help='Comma separated list of test datasets: ' +
+                         ' | '.join(_TEST_DATASET_NAMES) +
+                         ' (default: roxford5k,rparis6k).')
+flags.DEFINE_enum('precompute_whitening', default=None,
+                  enum_values=_PRECOMPUTE_WHITEN_NAMES,
+                  help='Dataset used to learn whitening: ' +
+                       ' | '.join(_PRECOMPUTE_WHITEN_NAMES) +
+                       ' (default: None).')
+flags.DEFINE_integer('test_freq', default=5,
+                     help='Run test evaluation every N epochs (default: 1).')
+flags.DEFINE_string('multiscale', default='[1.]',
```

Comment:
I think in this case the examples you give in the `help` section of the flag would need to be changed, since now it should be input as a comma-separated string?